### DISCUSSION TITLE
Why we are not using any PR template?

### DISCUSSION THREAD

**Discussion by mikkokotila** [world-federation-of-advertisers/cross-media-measurement#1898] [Ideas & Requests]
Is there a particular reason why we are not using any kind of PR template? The content could be as simple as having the author check (or leave unchecked) few statements we deem to be important.

**Comment by kungfucraig** [comment]
What problems would using a template solve? 

Also note that we already use conventional commits. See: https://www.conventionalcommits.org/en/v1.0.0/

Also see our dev standards document for guidance on PR messages: https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/main/docs/dev-standards.md

**Comment by mikkokotila** [comment]
PR template is a sure way to help new contributors to follow the conventions. For example, we have odd cases where the name of the PR starts in a non-standard way, or does not apply a name (e.g. fix) at all. While this is minor in our case, it is an unnecessary leak. 

So for example, we could start by adding a single checkbox with a truth statement `The PR is named as per the convention` where convention is a link to the underlying convention doc. That is an example of an immediate specific case. 

Generally, I find that having a PR template is a good way to have the engineer stop, and ask a few key questions from themselves. Truth statements and checkboxes are a wonderful way to achieve this.

**Reply by SanjayVas** [reply]
One issue is that we use the pull request body as part of the commit message. Therefore we do not want anything extraneous in the PR body. My understanding is that PR templates work by adding text into the PR body.

As for enforcing Conventional Commits, the commitlint tool can be run in a GitHub Actions PR check. I just haven't gotten around to integrating it yet.

**Reply by mikkokotila** [reply]
Excellent. Case closed. 

Also, really great to see we are adhering to Conventional Commits to the degree we are.
